src/main/kotlin/org/jetbrains/teamcity/rest/api.kt

```diff
@@ -33,6 +33,8 @@ interface VcsRootLocator {
 interface BuildLocator {
     fun fromConfiguration(buildConfigurationId: BuildConfigurationId): BuildLocator
 
+    fun withNumber(buildNumber: String): BuildLocator
```

Comment:
I need this because I'm searching build with exact buildNumber in different buildConfigurations. Otherwise I have to use those mentioned method and wrap it inside try-catch (if there is no build with provided buildNumber inside buildConfiguration) which is not good I think. But with withNumber I can do several requests and provide a non-null check and that's it.